app/Platform/Actions/ComputeGameSearchTitlesAction.php

```diff
@@ -0,0 +1,387 @@
+        foreach ($seriesAbbreviationMap as $seriesName => $seriesAbbreviations) {
+            if (stripos($gameName, $seriesName) !== false) {
+                foreach ($seriesAbbreviations as $abbreviation) {
+                    $abbreviations[] = mb_strtolower($abbreviation . $number);
+                }
+            }
+        }
+    }
+
+    /**
+     * Extract series names from game titles to improve search relevance.
+     *
+     * This method identifies and extracts base series names from full game titles,
+     * allowing users to find games by searching just the series name. For example,
+     * users often search for "zelda" instead of "The Legend of Zelda", or "final fantasy"
+     * instead of the full title. By explicitly adding these series names as search terms,
+     * we ensure better search results when users search by series.
+     */
+    private function extractSeriesNames(string $title): array
+    {
+        $seriesNames = [];
+
+        // Map of title patterns to their common search terms.
+        // This was picked very naively - we can expand this at any time.
+        $seriesPatternMap = [
+            '/^(Assassin\'s Creed)\s*/i' => 'assassins creed',
+            '/^(Call of Duty)\s*/i' => 'call of duty',
+            '/^(Castlevania)\s*/i' => 'castlevania',
+            '/^(Devil May Cry)\s*/i' => 'devil may cry',
+            '/^(Donkey Kong)\s*/i' => 'donkey kong',
+            '/^(Dragon Quest)\s*/i' => 'dragon quest',
+            '/^(Dragon Warrior)\s*/i' => 'dragon warrior',
+            '/^(Elder Scrolls)\s*/i' => 'elder scrolls',
+            '/^(F-Zero)\s*/i' => 'f-zero',
+            '/^(Final Fantasy)\s*/i' => 'final fantasy',
+            '/^(Fire Emblem)\s*/i' => 'fire emblem',
+            '/^(God of War)\s*/i' => 'god of war',
+            '/^(Grand Theft Auto)\s*/i' => 'grand theft auto',
+            '/^(Halo)\s*/i' => 'halo',
+            '/^(Kingdom Hearts)\s*/i' => 'kingdom hearts',
+            '/^(Kirby)\s*/i' => 'kirby',
+            '/^(Legend of Zelda)[:;]?\s*/i' => 'zelda',
+            '/^(Mario Kart)\s*/i' => 'mario kart',
+            '/^(Mario Party)\s*/i' => 'mario party',
+            '/^(Mega Man)\s*/i' => 'mega man',
+            '/^(Metal Gear)\s*/i' => 'metal gear',
+            '/^(Metroid)\s*/i' => 'metroid',
+            '/^(Mortal Kombat)\s*/i' => 'mortal kombat',
+            '/^(Pokemon|Pokémon)\s*/i' => 'pokemon',
+            '/^(Resident Evil)\s*/i' => 'resident evil',
+            '/^(Silent Hill)\s*/i' => 'silent hill',
+            '/^(Sonic the Hedgehog)\s*/i' => 'sonic',
+            '/^(Sonic)\s*/i' => 'sonic',
+            '/^(Star Fox)\s*/i' => 'star fox',
+            '/^(Street Fighter)\s*/i' => 'street fighter',
+            '/^(Super Mario)\s*/i' => 'super mario',
+            '/^(The Elder Scrolls)\s*/i' => 'elder scrolls',
+            '/^(The Legend of Zelda)[:;]?\s*/i' => 'zelda',
+            '/^(Zelda)\s*/i' => 'zelda',
+        ];
```

Comment:
What is the point of this function? Other than "The Legend of Zelda" being mapped to "Zelda", and the accent removal in Pokémon (which I expect meleisearch to do for us), it seems to just be detecting known values and then lowercasing them. Aren't the case-normalized words already going to be in the search terms extracted by meleisearch?